processor/transformprocessor/internal/metrics/func_extract_sum_metric.go

```diff
@@ -0,0 +1,117 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottldatapoint"
+)
+
+type extractSumMetricArguments struct {
+	StringAggTemp string `ottlarg:"0"`
```

Comment:
The `convert_summary_*` functions were created before we had ENUMs and going back to update the functions was a breaking change so we never did it.  Since we are introducing a new function to handle all types, though, how do you feel about aggregationTemporality being an ENUM instead of a string?

Reply:
Like this: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24368/commits/39da6ad5ad676bbb1ea8be63bdd3e299958e8632? I'm not completely sure if I need to do the conversion between `ottl.Enum` and `pmetric.AggregationTemporality` manually, or if there's some magic that will do it for me if I declare the args struct correctly.

Reply:
That looks correct. You do  have to do the conversion.